src/Uno.UI.Runtime.Skia.WebAssembly.Browser/Rendering/BrowserRenderer.cs

```diff
@@ -60,18 +63,48 @@ private void Initialize()
 		_jsInfo = NativeMethods.CreateContext(this, _nativeSwapChainPanel, WebAssemblyWindowWrapper.Instance?.CanvasId ?? "invalid");
 	}
 
-	internal void InvalidateRender() => NativeMethods.Invalidate(_nativeSwapChainPanel);
-
-	private void RenderFrame()
+	internal void InvalidateRender()
 	{
-		using var _ = _fpsHelper.BeginFrame();
-
 		if (_host.RootElement is { } rootElement && (rootElement.IsArrangeDirtyOrArrangeDirtyPath || rootElement.IsMeasureDirtyOrMeasureDirtyPath))
```

Comment:
Having the dirty check inside InvalidateRender and returning early without forcing a later invalidation can lead to the UI not updating.
Here's a scenario
1. 2 UIElements UI1 and UI2 are dirty
2. UI1 updated its layout and in the process called InvalidateRender, which did nothing because UI2 is still dirty.
3. UI2 goes through its layout cycle but nothing is updated and therefore no new InvalidateRender call happens.
4. Result: UI1 changed dimensions but the UI was never updated.